services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,17 +43,46 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  if (thermalMgrQueueHandle == NULL) return ERR_CODE_INVALID_STATE;
+  if (xQueueSend(thermalMgrQueueHandle, event, 0) == errQUEUE_FULL) return ERR_CODE_QUEUE_FULL;
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t event;
+
+  event.type = THERMAL_MGR_EVENT_HANDLE_INTERRUPT;
+  thermalMgrSendEvent(&event);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
   while (1) {
+    thermal_mgr_event_t buffer;
+    if(xQueueReceive(thermalMgrQueueHandle, &(buffer), 0) == pdPASS){
```

Comment:
I would make this a guard clause. So negate the condition. Move everything out and replace it with a continue statement